dynamiqs/core/propagator_solver.py

```diff
@@ -0,0 +1,57 @@
+from __future__ import annotations
+
+from abc import abstractmethod
+
+import jax
+import jax.numpy as jnp
+from jax import Array
+from jaxtyping import PyTree, Scalar
+
+from ..time_array import ConstantTimeArray
+from .abstract_solver import BaseSolver, MESolver
+
+
+class PropagatorSolver(BaseSolver):
+    def __init__(self, *args):
+        super().__init__(*args)
+        # check that Hamiltonian is time-independent
+        if not isinstance(self.H, ConstantTimeArray):
+            raise TypeError(
+                'Solver `Propagator` requires a time-independent Hamiltonian.'
+            )
+        self.H = self.H.x
+
+    def run(self) -> PyTree:
+        # === solve differential equation
+        def f(y, delta_t):
```

Comment:
```suggestion
        def propagate(y, delta_t):
```
You seem to like naming everything `f` these days, but it's not very explicit :p